rekallpy/rekall/interval_set.py

```diff
@@ -809,32 +810,63 @@ def coalesce(self,
         Returns:
             A new IntervalSet of intervals that are disjoint along ``axis`` and
             are at least ``epsilon`` apart.
+            
         """
-
-        def update_output(output, intrvl):
-            if len(output) == 0:
-                output.append(intrvl)
-            else:
-                merge_candidate = output[-1]
+        if (len(self._intrvls) == 0):
+            return self
+
+        new_coalesced_intrvls = []
+        current_intrvls = []
+
+        for intrvl in self._intrvls:
```

Comment:
You'll want to loop through a version of `self._intrvls` that's sorted by `axis` first, otherwise this algorithm won't work (I know it didn't work before, but we may as well fix it while we're at it.

You should make a copy of `self._intrvls` and sort that.